### ISSUE TITLE
Parsing drops extensions on primitives

### ISSUE THREAD

**Issue by jmandel** [smart-on-fhir/client-py#30]
Documenting a known issue: parsing FHIR resources with extensions on primitives silently drops the extensions. For example:

```
{
  "resourceType": "Patient",
  "gender": "male",
  "_gender": {
    "extension": [{
      "url": "http://test-extension",
      "valueString": "with extra data"
    }]
  }
}
```

**Comment by p2** [comment]
Yes, thanks! Also to have it documented: the Swift-FHIR framework now solves this by providing custom classes/structs for primitives. Those can handle extensions (and ids), as opposed to the native strings/numbers used by the Python client.

**Comment by dajaffe** [comment]
Hi - I just ran into this trying to return back resources which are using string extensions for language translations and attempting to `as_json` the resource, i.e.:

```
"item": [
  "text": "hello",
  "_text": {
    "extension": [
      {
        "url": "http://hl7.org/fhir/StructureDefinition/translation",
         "extension": [
           {
             "url": "lang",
             "valueCode": "fr"
           },
           {
             "url": "content",
             "valueString": "bonjour"
           }
         ]
      }
    ]
  }
]
```

Is being able to support this planned in the future?

**Comment by mikix** [comment]
It is definitely something we ought to do. And as Pascal says above, one reasonable approach would be custom types. Something as simple as the following might work:

```
class StringType:
  value = None
  extensions = None
```

But... that's a big change, and a breaking one. There are also backwards-compatible options, like adding underscored fields that shadow all the primitives in the model classes.

I dunno, I haven't thought about it much yet.

But it's a direction we'd want to go and worth doing. Patches appreciated if folks have thoughts / time for it. But note that patches like that should be done in [fhir-parser](https://github.com/smart-on-fhir/fhir-parser/) first, then ported here.

**Comment by p2-apple** [comment]
This should be possible without changes to fhir-parser, just updates to the Python-specific templates like [fhir-parser-resources/template-resource.py](https://github.com/smart-on-fhir/client-py/blob/main/fhir-parser-resources/template-resource.py)

**Comment by mikix** [comment]
Yes it's possible for sure. But if we can keep the two repos in rough sync, that'd be nice - since this change isn't specific to `fhirclient`, but all consumers of `fhir-parser` would appreciate the improvement.

**Comment by p2-apple** [comment]
We already do support extensions on primitive types in https://github.com/apple/fhirmodels, with the current version of fhir-parser.
